src/main/java/net/openhft/chronicle/queue/rollcycles/RollCycleArithmetic.java

```diff
@@ -0,0 +1,53 @@
+package net.openhft.chronicle.queue.rollcycles;
+
+import net.openhft.chronicle.core.Maths;
+import net.openhft.chronicle.queue.RollCycles;
+
+import static net.openhft.chronicle.queue.RollCycle.MAX_INDEX_COUNT;
+
+/**
+ * Reusable optimised arithmetic for converting between cycles/indices/sequenceNumbers
+ */
+public final class RollCycleArithmetic {
```

Comment:
This is how we make the arithmetic reusable, it should get in-lined trivially because it's a final class?